cf-agent/verify_files.c

```diff
@@ -416,7 +416,7 @@ static PromiseResult VerifyFilePromise(EvalContext *ctx, char *path, const Promi
 /* Phase 1 - */
 
     if ((exists && ((a.haverename) || (a.haveperms) || (a.havechange) || (a.transformer))) ||
-        ((exists || link) && a.havedelete))
+        ((exists || link) && a.havedelete) ||  (a.acl.acl_entries != NULL))
```

Comment:
What if the file doesn't exist? I think the check would work better inside the first set of parentheses, after the `if exists &&` clause.